packages/xl-ai/src/AIExtension.ts

```diff
@@ -134,6 +137,51 @@ export class AIExtension extends BlockNoteExtension {
         options.agentCursor || { name: "AI", color: "#8bc6ff" },
       ),
     );
+
+    // Scrolls to the block being edited by the AI while auto scrolling is
+    // enabled.
+    this.editor.onCreate(() => {
+      this.editor.onChange(() => {
+        if (!this.autoScroll) {
+          return;
+        }
+
+        const aiMenuState = this._store.getState().aiMenuState;
+        const aiMenuNonErrorState =
+          aiMenuState === "closed" ? undefined : aiMenuState;
+        if (aiMenuNonErrorState?.status === "ai-writing") {
+          const blockElement = this.editor.domElement?.querySelector(
+            `[data-node-type="blockContainer"][data-id="${aiMenuNonErrorState.blockId}"]`,
+          );
+          blockElement?.scrollIntoView({ block: "center" });
+        }
+      });
+    });
```

Comment:
Why wrap in an `editor.onCreate`? I'm actually uncertain that `onCreate` gets invoked right now, does it? Is it on mount?

Reply:
We need to fix this then, because I don't think this is a great pattern